doc/source/development/rfc/rfc85_policy_code_additions.rst

```diff
@@ -0,0 +1,85 @@
+Started: 2022-01-17
+Status:  Development
+======== ==================================================
+
+Summary
+-------
+
+This document describes the policies that the GDAL project will apply to assess
+substantial code additions, typically new drivers, in particular coming for new
+contributors to the project.
+
+Motivation
+----------
+
+The GDAL project has historically been quite open to new code additions, including
+drivers that rely on proprietary licensed and/or closed-source SDKs (called
+"binary SDK" in the rest of this document). This approach is part
+of its strengths, but also comes at a price, with the software being larger
+and larger, with sometimes contributions being abandoned by their authors without
+new maintainers.
+
+Being a free & open-source project, the GDAL project will apply stronger scrutiny to
+code that require a binary SDK and encourage contributors to submit code additions
+that do not depend on such binary SDK or license such SDK under free & open-source
+license terms.
+
+Policy
+------
+
+- Drivers require a designated responsible contact, tracked in
+  https://github.com/OSGeo/gdal/wiki/Maintainers-per-sub-system
+
+- Contributions should follow other RFCs describing development rules, and come
+  with test scripts and sufficient documentation, covering usage and build instructions.
+
+- If the driver require a binary SDK not downloadable without cost, or that requires
+  a complicated registration process, the GDAL team is unlikely to support
+  driver inclusion.
+
+- If the binary SDK is no longer current, or modernized to work with latest
+  compilers and GDAL, the driver can be dropped. This rule also applies to open-source
+  dependencies that are no longer maintained or are superseded by other alternatives.
+
+- If the driver has unaddressed bugs, is breaking continuous integration (CI),
+  has caused CI bits to be disabled, or has not caught up to API modifications
+  requiring updates (extremely rare), the driver will be dropped from build
+  scripts, and thus will not be built without significant user intervention.
+
+- If critical/blocking issues reported in a driver are not addressed within a
+  2 month time-frame, it can be dropped from the tree entirely by a designated
+  GDAL maintainer for all releases going forward.
+
+- Contributors of significant code additions are expected to participate to the
+  day-to-day life of the project, and need to monitor closely the communication
+  channels of the project: issue tracker, mailing lists, etc. They are expected at
+  a minimum to respond in a timely manner to changes that affect the whole project:
+  CI, build scripts,upgrade of dependencies & build tools, documentation etc.
+  They are also expected to help sharing the burden of maintenance and participate
+  to the global health of the project by doing pull request reviews, tackling general
+  bug reports, functional enhancements, documentation and infrastructure enhancements,
+  etc.
```

Comment:
If someone is a dedicated & diligent maintainer for the WizBang driver, do we really _expect_ them to also become a maintainer of the entire GDAL project?

Reply:
I do not read that so that they are expected to maintain the _entire_ project. Rather to do a little bit extra to keep the GDAL ship that is carrying their load afloat.

Reply:
> I do not read that so that they are expected to maintain the entire project. Rather to do a little bit extra to keep the GDAL ship that is carrying their load afloat.

Exactly. The project can't work if people wear blinders and only care about the part that they contributed to without considering the project as a whole

Reply:
@jratike80 that's exactly my general expectation of what we would want, but that's not how it read to me. I've added some different wording below as a suggestion.